### PR TITLE
added debounce on sortBy

### PR DESCRIPTION
**Pull request by samouzegar10**

### CODE REVIEW

**Comment by samouzegar10** [comment]
the current fix aborts any ongoing api call. 
aborting the last ongoing is functional.
in samples page, the same initial call happens twice at the beginning for some reason.

File: frontend/src/components/filters/filterComponents/DebouncedInput.jsx
Comment (nafiz1001) [inline frontend/src/components/filters/filterComponents/DebouncedInput.jsx:1]: This your opportunity to convert this to typescript 😉

File: frontend/src/modules/samples/actions.ts
Comment (nafiz1001) [inline frontend/src/modules/samples/actions.ts:1]: This seems like an unnecessary change, but I'd let it pass since it's very small.

File: frontend/src/utils/api.ts
Comment (nafiz1001) [inline frontend/src/utils/api.ts:1]: This is not your code but I find it strange that we delete an entry in `ongoingRequests` when we want to abort. I feel like it should be the opposite, but I'd verify beforehand.

```ts
        return request.then(res => {
            if (options.abort) {
                delete ongoingRequests[baseRoute]
            }
            return res
})
```

Comment (samouzegar10) [reply frontend/src/utils/api.ts:1]: when you say, it should be the opposite, can you elaborate?

Comment (nafiz1001) [reply frontend/src/utils/api.ts:1]: Oh wait. I see now that the callback only gets executed after the request has been resolved. So this is just cleaning up the entry for later request.

Comment (samouzegar10) [reply frontend/src/utils/api.ts:1]: indeed, firstly the abortController does its thing (assign the signal, right now theres an empty message and then it executes abort(), this ongoingRequests[baseRoute] is a "global" variable used to keep track.

File: frontend/src/components/filters/filterComponents/DebouncedInput.jsx
Comment (samouzegar10) [reply frontend/src/components/filters/filterComponents/DebouncedInput.jsx:1]: the change was very minimal and the app seems to be working as if nothing changed

Comment (nafiz1001) [reply frontend/src/components/filters/filterComponents/DebouncedInput.jsx:1]: I didn't mean just change file. I meant also adding adjusting types here and there. Just enough to eliminate any `any`s and type errors.

Comment (samouzegar10) [reply frontend/src/components/filters/filterComponents/DebouncedInput.jsx:1]: ah, you had a certain view of what to do. 
want to have a conversation about it tomorrow :) ?

Comment (nafiz1001) [reply frontend/src/components/filters/filterComponents/DebouncedInput.jsx:1]: Sure

File: frontend/src/modules/samples/actions.ts [hunk 1]
```diff
@@ -39,8 +39,10 @@ export const update = (id, sample) => async (dispatch, getState) => {
 export const list = (options) => async (dispatch: AppDispatch, getState) => {
     const params = { limit: 100000, ...options }
     return await dispatch(networkAction(LIST,
-        api.samples.list(params),
-        { meta: params }
+        api.samples.list(params, { abort: true }),
+        {
+            meta: params,
+        }
```
Comment (nafiz1001) [inline frontend/src/modules/samples/actions.ts:45]: Now that I think about it. This might be risky because there might be different React components that depends on sample list API call. So if multiple components fire this API at once, one of them will be aborted which is no good. But I don't know any page on top of my head that could cause this.

File: frontend/src/components/pagedItemsTable/PagedItemsTable.tsx [hunk 2]
```diff
@@ -251,7 +247,7 @@ function PagedItemsTable<T extends object>({
 						columns={columns}
 						rowKey={getRowKeyForDataObject}
 						scroll={{x: 300}}
-						onChange={sortByCallback}
+						onChange={useDebounce(sortByCallback)}
```
Comment (nafiz1001) [inline frontend/src/components/pagedItemsTable/PagedItemsTable.tsx:250]: Be careful about conditional rendering. `useDebounce` is a hook which must always be executed for each re-rendering.

File: frontend/src/modules/samples/actions.ts (hunk 1 above)
Comment (samouzegar10) [reply frontend/src/modules/samples/actions.ts:45]: it would have been risky, although, its pin pointed down to the samples list content page (table) even if lets say, another page would use it, the call itself will be safe until he makes a 2nd call on the same api call. its pretty safe as is :) im testing locally to see if anything changes in the app and i havent seen any changes in behavior

File: frontend/src/components/pagedItemsTable/PagedItemsTable.tsx (hunk 2 above)
Comment (samouzegar10) [reply frontend/src/components/pagedItemsTable/PagedItemsTable.tsx:250]: sure no problem! this is sound and clear

Comment (samouzegar10) [reply frontend/src/components/pagedItemsTable/PagedItemsTable.tsx:250]: no major changes either, all is working as it should.

File: frontend/src/utils/api.ts [hunk 3]
```diff
@@ -362,7 +362,7 @@ function apiFetch<R extends ResponseWithData<any>>(method: HTTPMethod, route: st
             const controller = new AbortController()
             signal = controller.signal
             if (ongoingRequests[baseRoute]) {
-                ongoingRequests[baseRoute].abort("")
+                ongoingRequests[baseRoute].abort("Aborting previous unfinished api call.")
```
Comment (nafiz1001) [inline frontend/src/utils/api.ts:365]: It might be more helpful to include the API route that failed such as the `baseRoute`.

Comment (samouzegar10) [reply frontend/src/utils/api.ts:365]: thats a good idea :)

**Comment by UlysseFG** [comment]
I think i like more the useDebounce version.

I do get this notification message though sometime when changing filter or changing sort fast :

action.error.message is undefined

./src/store.ts/notificationError/</<@http://localhost:9000/main.js?8199501887c1e93c9bb4:45001:11
./node_modules/redux-logger/dist/redux-logger.js/</S/</</<@http://localhost:9000/vendors-node_modules_ant-design_icons_es_icons_ArrowLeftOutlined_js-node_modules_ant-design_i-ed462e.js?8199501887c1e93c9bb4:215478:7667
./node_modules/redux-thunk/es/index.js/createThunkMiddleware/middleware/</<@http://localhost:9000/vendors-node_modules_ant-design_icons_es_icons_ArrowLeftOutlined_js-node_modules_ant-design_i-ed462e.js?8199501887c1e93c9bb4:216596:16
dispatch@http://localhost:9000/vendors-node_modules_ant-design_icons_es_icons_ArrowLeftOutlined_js-node_modules_ant-design_i-ed462e.js?8199501887c1e93c9bb4:217320:28
./src/utils/actions.ts/networkAction/</<@http://localhost:9000/main.js?8199501887c1e93c9bb4:45078:15

**Comment by nafiz1001** [comment]
> I think i like more the useDebounce version.
> 
> I do get this notification message though sometime when changing filter or changing sort fast :
> 
> action.error.message is undefined
> 
> ./src/store.ts/notificationError/</<@http://localhost:9000/main.js?8199501887c1e93c9bb4:45001:11 ./node_modules/redux-logger/dist/redux-logger.js/</S/</</<@http://localhost:9000/vendors-node_modules_ant-design_icons_es_icons_ArrowLeftOutlined_js-node_modules_ant-design_i-ed462e.js?8199501887c1e93c9bb4:215478:7667 ./node_modules/redux-thunk/es/index.js/createThunkMiddleware/middleware/</<@http://localhost:9000/vendors-node_modules_ant-design_icons_es_icons_ArrowLeftOutlined_js-node_modules_ant-design_i-ed462e.js?8199501887c1e93c9bb4:216596:16 dispatch@http://localhost:9000/vendors-node_modules_ant-design_icons_es_icons_ArrowLeftOutlined_js-node_modules_ant-design_i-ed462e.js?8199501887c1e93c9bb4:217320:28 ./src/utils/actions.ts/networkAction/</<@http://localhost:9000/main.js?8199501887c1e93c9bb4:45078:15

I sometimes come across this sort of error. For me it's often because I called `dispatch` on something that's neither a redux thunk nor an object that describes a redux action.

**Comment by nafiz1001** [comment]
I haven't checked the no `useDebounce` option yet.

**Comment by nafiz1001** [comment]
> I haven't checked the no `useDebounce` option yet.

debounce is totally worth it even locally
